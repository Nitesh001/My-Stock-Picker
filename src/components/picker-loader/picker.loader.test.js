import React from 'react';
import renderer from 'react-test-renderer';
import PickerLoader from './picker.loader';

test('Loader shows default message', () => {
  const component = renderer.create(
    <PickerLoader />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});

test('Loader shows custom message', () => {
  const component = renderer.create(
    <PickerLoader mssg="custom message" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})