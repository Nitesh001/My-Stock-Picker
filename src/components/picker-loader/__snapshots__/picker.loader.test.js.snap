// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader shows custom message 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rr-ui-event-key={null}
    onClick={[Function]}
  >
    custom message
  </div>
</div>
`;

exports[`Loader shows default message 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rr-ui-event-key={null}
    onClick={[Function]}
  >
    loading...
  </div>
</div>
`;
